packages/amplify-cli/src/error-handler/amplify-error-handler.ts

```diff
@@ -0,0 +1,101 @@
+import { $TSAny, AmplifyError, AmplifyErrorType } from 'amplify-cli-core';
+import { printer } from 'amplify-prompts';
+import { Context } from '../domain/context';
+import { reportError } from '../commands/diagnose';
+import { isHeadlessCommand } from '../context-manager';
+
+let context: Context;
+
+/**
+ * Inject a reference to the context object into this module
+ */
+export const init = (_context: Context): void => {
+  context = _context;
+};
+
+/**
+ * Handle fatal errors
+ */
+export const handleError = async (err: unknown): Promise<void> => {
+  let amplifyError: AmplifyError;
+
+  if (err instanceof AmplifyError) {
+    amplifyError = err;
+  } else if (!(err instanceof Error)) {
+    amplifyError = unknownErrorToAmplifyError(err);
+  } else if (isNodeJsError(err)) {
+    amplifyError = nodeErrorToAmplifyError(err);
+  } else {
+    amplifyError = genericErrorToAmplifyError(err);
+  }
+
+  if (context?.usageData) {
+    context?.usageData.emitError(amplifyError);
+  }
+
+  if (context && isHeadlessCommand(context)) {
+    printHeadlessAmplifyError(amplifyError);
+  } else {
+    printAmplifyError(amplifyError);
+  }
+
+  if (context) {
+    await reportError(context, amplifyError);
+  }
+};
+
+const printAmplifyError = (amplifyError: AmplifyError): void => {
+  const {
+    message, details, resolution, link, stack,
+  } = amplifyError;
+
+  printer.error(message);
+  if (details) {
+    printer.info(details);
+  }
+  printer.blankLine();
+  printer.info(`Resolution: ${resolution}`);
+  if (link) {
+    printer.info(`Learn more at: ${link}`);
+  }
+  printer.blankLine();
+  if (stack) {
+    printer.debug(stack);
+  }
+  process.exitCode = 1;
+};
+
+const printHeadlessAmplifyError = (amplifyError: AmplifyError): void => {
+  printer.error(amplifyError.toJson());
```

Comment:
this needs to print to stderr instead of stdout. You can construct a printer that writes to stderr using `errPrinter = new AmplifyPrinter(process.stderr)`